modules/lib/lib-event/src/main/resources/lib/xp/event.ts

```diff
@@ -0,0 +1,143 @@
+declare global {
+    interface XpLibraries {
+        '/lib/xp/event': typeof import('./event');
+    }
+}
+
+export interface ListenerParams<EventData extends object = EnonicEventData> {
+    /**
+     * Event type pattern
+     */
+    type: EnonicEventTypes;
+
+    /**
+     * Callback event listener
+     */
+    callback: (event: EnonicEvent<EventData>) => void;
+
+    /**
+     * Local events only (default to false)
+     */
+    localOnly?: boolean;
+}
+
+export interface SendParams {
+    /**
+     * Event type
+     */
+    type: EnonicEventTypes;
+
+    /**
+     * `true` if it should be distributed in cluster
+     */
+    distributed?: boolean;
+
+    /**
+     * Additional data for event.
+     */
+    data?: object;
+}
+
+export interface EnonicEvent<EventData extends object = EnonicEventData> {
+    readonly type: EnonicEventTypes | string;
+    readonly timestamp: number;
+    readonly localOrigin: boolean;
+    readonly distributed: boolean;
+    readonly data: EventData;
+}
+
+export interface EnonicEventData {
+    readonly nodes: ReadonlyArray<EnonicEventDataNode>;
+    readonly state?: string; // event type <'node.stateUpdated'>
+}
+
+export interface EnonicEventDataNode {
+    readonly id: string;
+    readonly path: string;
+    readonly branch: string;
+    readonly repo: string;
+    readonly newPath?: string; // event type <'node.moved' | 'node.renamed'>
+}
+
+type LiteralUnion<T extends U, U = string> = T | (U & Record<never, never>);
+
+export type EnonicEventTypes =
+    LiteralUnion<
+    | 'node.*'
+    | 'node.created'
+    | 'node.deleted'
+    | 'node.pushed'
+    | 'node.duplicated'
+    | 'node.updated'
+    | 'node.moved'
+    | 'node.renamed'
+    | 'node.sorted'
+    | 'node.stateUpdated'
+    | 'node.permissionsUpdated'
+    >;
+
+interface EventSenderHelper {
+    setType(type: string): void;
+    setDistributed(distributed: boolean): void;
+    setData(value: object): void;
+    send(): void;
+}
+
+interface EventListenerHelper {
+    setType(type: string): void;
+    setListener<T = unknown>(callback: (event: T) => void): void;
+    setLocalOnly(localOnly: boolean): void;
+    register(): void;
+}
+
+/**
```

Comment:
where do the jsdock land now?

Reply:
I may have forgot to test it. Can you point me what should be generated and what tasks do we use for that?